app/services/reservation_creator.rb

```diff
@@ -0,0 +1,92 @@
+class ReservationCreator
+
+  attr_reader :order, :order_detail, :params, :error
+
+  delegate :merged_order?, :instrument_only_order?, to: :status_q
+
+  def initialize(order, order_detail, params)
+    @order = order
+    @order_detail = order_detail
+    @params = params
+  end
+
+  def save(session_user)
+    if !@order_detail.bundled? && params[:order_account].blank?
+      @error = I18n.t("controllers.reservations.create.no_selection")
+      return false
+    end
+
+    Reservation.transaction do
+      begin
+        update_order_account
+
+        # merge state can change after call to #save! due to OrderDetailObserver#before_save
+        to_be_merged = @order_detail.order.to_be_merged?
+
+        save_reservation_and_order_detail(session_user)
+
+        if to_be_merged
+          # The purchase_order_path or cart_path will handle the backdating, but we need
+          # to do this here for merged reservations.
+          backdate_reservation_if_necessary(session_user)
+          @success = :merged_order
+        elsif @order.order_details.one?
+          @success = :instrument_only_order
+        else
+          @success = :default
+        end
+      rescue ActiveRecord::RecordInvalid => e
+        @error = e.message
+        raise ActiveRecord::Rollback
+      rescue => e
+        @error = I18n.t("orders.purchase.error")
```

Comment:
Minor: since there are no tests for this class directly I checked code coverage. This `rescue` block looks like the only part of the class that isn't getting tested.